viewer/src/datamodels/cad/sector/CachedRepository.ts

```diff
@@ -54,200 +68,183 @@ export class CachedRepository implements Repository {
   }
 
-  getLoadingStateObserver(): Observable<boolean> {
-    return this._isLoadingSubject.pipe(distinctUntilChanged(), share());
+  private loadSimpleAndDetailedSector(): OperatorFunction<WantedSector, ConsumedSector> {
+    return publish(wantedSectorObservable => {
+      const [cachedSectorObservable, uncachedSectorObservable] = partition(wantedSectorObservable, wantedSector =>
+        this._modelDataCache.has(this.cacheKey(wantedSector))
+      );
+
+      return merge(
+        cachedSectorObservable.pipe(flatMap(wantedSector => this._modelDataCache.get(this.cacheKey(wantedSector)))),
+        uncachedSectorObservable.pipe(this.loadSimpleAndDetailedSectorFromNetwork())
+      );
+    });
   }
 
-  clearCache() {
-    this._modelDataCache.clear();
+  private loadSimpleAndDetailedSectorFromNetwork(): OperatorFunction<WantedSector, ConsumedSector> {
+    return flatMap(wantedSector => {
+      if (wantedSector.levelOfDetail === LevelOfDetail.Simple) {
+        return this.loadSimpleSectorFromNetwork(wantedSector);
+      } else if (wantedSector.levelOfDetail === LevelOfDetail.Detailed) {
+        return this.loadDetailedSectorFromNetwork(wantedSector);
+      } else {
+        throw new Error('Unhandled LevelOfDetail');
+      }
+    }, this._concurrentNetworkOperations);
   }
 
-  getParsedData(): Observable<{ blobUrl: string; lod: string; data: SectorGeometry | SectorQuads }> {
-    return this._parsedDataSubject.pipe(distinct(keySelector => '' + keySelector.blobUrl + '.' + keySelector.sectorId)); // TODO: Should we do replay subject here instead of variable type?
+  private loadSimpleSectorFromNetwork(wantedSector: WantedSector): Observable<ConsumedSector> {
+    const networkObservable: Observable<ConsumedSector> = onErrorResumeNext(
+      from(
+        this._modelSectorProvider.getCadSectorFile(wantedSector.blobUrl, wantedSector.metadata.facesFile.fileName!)
+      ).pipe(
+        catchError(error => {
+          // tslint:disable-next-line: no-console
+          console.error('loadSimple request', error);
+          this._modelDataCache.remove(this.cacheKey(wantedSector));
+          throw error;
+        }),
+        flatMap(buffer => this._modelDataParser.parseF3D(new Uint8Array(buffer))),
+        tap(sectorQuads => {
+          this._parsedDataSubject.next({
+            blobUrl: wantedSector.blobUrl,
+            sectorId: wantedSector.metadata.id,
+            lod: 'simple',
+            data: sectorQuads
+          });
+        }),
+        map(sectorQuads => ({ ...wantedSector, data: sectorQuads })),
+        this._modelDataTransformer.transform(),
+        tap(group => {
+          group.name = `Quads ${wantedSector.metadata.id}`;
+        }),
+        map(group => ({ ...wantedSector, group })),
+        shareReplay(1),
+        take(1)
+      )
+    );
+    if (this._modelDataCache.isFull()) {
+      this._modelDataCache.cleanCache(10);
+    }
+    this._modelDataCache.add(this.cacheKey(wantedSector), networkObservable);
```

Comment:
I think `add` should free space itself and don't bother the caller to do it.

Reply:
I guess we could do that, but should it be part of this PR?

Reply:
As it's being changed now, I think so yes.

Reply:
I made a compromise

Reply:
What compromise was that? To me it looks identical still?

Reply:
I added two functions inside cached repository called `updateConsumedSectorCache` and `updateCtmFilesCache`. I prefer that to internalizing clean up, and even our test is based on the cache throw failing logic. If we are changing it to internally clean up then I want more than your and mine opinion on it, because I think that it sets a precedence on our coding guidelines